RFC-005-SWAP-Bitcoin-basic.md

```diff
@@ -0,0 +1,226 @@
+- How to construct a Hash Time Lock Contract (HTLC) to lock the Bitcoin asset on the Bitcoin blockchain.
+- How to deploy, redeem and refund the HTLC during the execution phase of RFC003
+
+
+## Motivation
+
+The motivation for creating this RFC is to allow implementations to negotiate and execute [RFC003 Basic HTLC Atomic Swaps](./RFC-003-SWAP-basic.md) using the COMIT protocol.
+The identity definition introduced in this RFC should also be useful for future protocols.
+
+## Bitcoin Identity
+
+The Identity to be used on Bitcoin is the 20-byte *pubkeyhash* which is the result of applying SHA-256 and then RIPEMD-160 to a user's compressed SECP256k1 public key.
+The compressed public key is used because it needs to be compatible with segwit transactions (see [BIP143](https://github.com/bitcoin/bips/blob/master/bip-0143.mediawiki#Restrictions_on_public_key_type)).
+
+While it may seem more intuitive to use a Bitcoin *address* as the identity, the pubkeyhash better fits the definition of identity given in RFC003. Using a Bitcoin address as the identity would require implementations to do a number of cumbersome validation steps such as such as verifying that it is a p2pkh or p2wpkh address, extraction of the pubkeyhash, validation of the network.
+
+When using the pubkeyhash as one of the identities in an RFC003 message body it MUST be encoded as 20 hex encoded string.
+This RFC extends the [registry](./registry.md) with the following entry in the identity table:
+
+| Ledger   | Identity Name | JSON Encoding            | Description                                                                                  |
+|:----     |:-------       |:-------------            | -------------------------------------------------------------------------------------------- |
+| Bitcoin  | `pubkeyhash`  | `hex-encoded-bytes (20)` | The result of applying SHA256 and then RIPEMD160 to a user's SECP256k1 compressed public key |
+
+## Hash Time Lock Contract
+
+### Hash Functions
+
+SHA-256 is the only value the `hash_function` header may take if Bitcoin is used as a ledger.
+
+### Parameters
+
+The parameters for the Bitcoin HTLC follow [RFC003](./RFC-003-SWAP-basic.md) and are described concretely in the following table:
+
+| Variable        | Description                                                                 |
+|:----------------|:----------------------------------------------------------------------------|
+| asset           | The quantity of satoshi                                                     |
+| secret_hash     | The hash of the `secret` (32 bytes for SHA-256)                             |
+| redeem_identity | The `pubkeyhash` requ the redeeming party                                   |
+| refund_identity | The `pubkeyhash` of the refunding party                                     |
+| expiry          | The absolute UNIX timestamp in seconds after which the HTLC can be refunded |
```

Comment:
I would like to add a note that implementations MUST check that this is a valid time to prevent against an attack where someone sends a very low value for expiry which looks is actually a absolute bock height rather than an unix epoch.

Reply:
I've added a note about this where we talk about `OP_CHECKLOCKTIMEVERIFY`.